### PR TITLE
feat: add Swagger 2.0 support with upgrade to OpenAPI 3.0 utility

### PR DESCRIPTION
**Pull request by TristanSpeakEasy**
# Add Swagger 2.0 Support with OpenAPI 3.0 Upgrade Utility

This PR adds comprehensive support for Swagger 2.0 specifications including full parsing, validation, walking, and an upgrade utility to convert to OpenAPI 3.0.

### Swagger 2.0 Support
- **Core models**: Complete Swagger 2.0 data models with marshalling support
- **Validation**: Full validation against Swagger 2.0 specification
- **Walking API**: Traverse Swagger documents with the same pattern as OpenAPI
- **Comprehensive tests**: 227 tests covering all Swagger functionality including round-trip testing with real-world specs (Petstore, DocuSign, Docker Engine API)

### Upgrade Utility
- **Automatic conversion**: Transform Swagger 2.0 to OpenAPI 3.0 with proper schema mapping
- **Key transformations**:
  - `swagger: "2.0"` → `openapi: "3.0.0"`
  - `host/basePath/schemes` → `servers`
  - `definitions` → `components.schemas`
  - `parameters` (body) → `components.requestBodies`
  - `securityDefinitions` → `components.securitySchemes`
  - Request/response schema wrapping in `content` objects
  - Reference rewriting (`#/definitions/` → `#/components/schemas/`)
- **17 upgrade tests** covering common and edge case scenarios

### CLI Commands
- `openapi swagger upgrade` - Convert Swagger 2.0 to OpenAPI 3.0
- `openapi swagger validate` - Validate Swagger 2.0 documents  

### Additional Improvements
- **JSON marshalling**: Fixed nul character handling to use proper JSON Unicode escapes (`\u0000`) instead of Go-style hex escapes (`\x00`)
- **Code quality**: Resolved all linting issues (gocritic, staticcheck, unparam, gosec, nilaway)
- **Test improvements**: All tests use `t.Context()` for better lifecycle management

## Testing
- ✅ All 227 swagger tests pass
- ✅ All 513 total tests pass (including JSON Schema test suite)
- ✅ Linting passes with 0 issues
- ✅ Round-trip testing with real-world Swagger specs validates correctness

## Breaking Changes
None - this is purely additive functionality.

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,9 +0,0 @@
-## Release Notes
```
Comment (BlakeTheAwesome) [inline RELEASE_NOTES.md:1]: I see this file was deleted - was it just checked in accidentally in PR #65?

Comment (TristanSpeakEasy) [reply RELEASE_NOTES.md:1]: Yeah basically tried to have copilot on the github website create a release and it instead created release notes

File: cmd/openapi/commands/swagger/root.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+// ReportValidationErrors reports validation errors if not writing to stdout
+func (p *SwaggerProcessor) ReportValidationErrors(validationErrors []error) {
+	if len(validationErrors) > 0 && !p.WriteToStdout {
+		fmt.Printf("⚠️  Found %d validation errors in original document:\n", len(validationErrors))
+		for i, validationErr := range validationErrors {
+			fmt.Printf("  %d. %s\n", i+1, validationErr.Error())
+		}
+		fmt.Println()
+	}
+}
```
Comment (BlakeTheAwesome) [inline cmd/openapi/commands/swagger/root.go:163]: In the `WriteToStdout` case here, can/should we be writing our errors to stderr instead?

Comment (TristanSpeakEasy) [reply cmd/openapi/commands/swagger/root.go:163]: probably, but I am planning on doing a separate pass on the CLI side of things to support the CLI interface better, handling std output better to allow pipelining, separate errors from normal output etc, clean up the ways arguments work. So I will tackle all that in a concerted effort separately

File: json/json.go [hunk 3]
```diff
@@ -1,141 +1,495 @@
...
+
+	// Build the indent string
+	indentStr := strings.Repeat(indent, indentCount)
+
+	// Create a JSON writer context
+	ctx := &jsonWriteContext{
+		indent:       indentStr,
+		buffer:       &bytes.Buffer{},
+		currentCol:   0,
+		forceCompact: indentCount == 0, // Force compact mode when no indentation
```
Comment (BlakeTheAwesome) [inline json/json.go]: I think this should probably be checking the length of `indentStr` instead, so that if the `indent` variable is an empty string it also turns on forceCompact.
It seems add to ever have a non-zero `indentCount` with an empty `indent` string, or to have an `indentCount` of zero with a non-empty `indent` string, but since the signature allows it we should probably make both work.

File: openapi/operation.go [hunk 4]
```diff
@@ -169,6 +169,7 @@ func (o *Operation) Validate(ctx context.Context, opts ...validation.Option) []e
 		errs = append(errs, securityRequirement.Validate(ctx, opts...)...)
 	}
 
+	// TODO allow validation of parameters, this isn't done at the moment as we would need to resolve references
```
Comment (BlakeTheAwesome) [inline openapi/operation.go:172]: Does this need to be done now or is it a future problem?

Comment (TristanSpeakEasy) [reply openapi/operation.go:172]: Future problem (discovered we aren't doing it in openapi) but basically I will fix it in a separate task to introduce linting to the library

File: yml/config.go [hunk 5]
```diff
@@ -99,18 +127,54 @@ func inspectData(data []byte) (OutputFormat, int) {
 			docFormat = OutputFormatJSON
 			foundDocFormat = true
 		default:
-			if len(line) != len(trimLine) && !foundIndentation {
-				indentation = len(line) - len(trimLine)
-				foundIndentation = true
+			currentLeading := len(line) - len(trimLine)
```
Comment (BlakeTheAwesome) [inline yml/config.go]: I think this calculation will be incorrect if `line` had a bunch of trailing whitespace.
```go
line := []byte("{    ")
trimLine := bytes.TrimSpace(line)
currentLeading := len(line) - len(trimLine)
fmt.Println(currentLeading) // Got 4, expected 0
```

**Review by BlakeTheAwesome — CHANGES_REQUESTED** [review]
Found a couple of minor issues.
I didn't go into detail reading all the swagger types or the tests, but I assume the tests should cover that stuff pretty well.

File: cmd/openapi/commands/swagger/root.go (hunk 2 above)
Comment (BlakeTheAwesome) [reply cmd/openapi/commands/swagger/root.go:163]: Fair enough. Is there a ticket for that work somewhere? It seems worth having written down so it doesn't get lost at any point.

Comment (TristanSpeakEasy) [reply cmd/openapi/commands/swagger/root.go:163]: https://github.com/speakeasy-api/openapi/discussions/67

File: openapi/operation.go (hunk 4 above)
Comment (TristanSpeakEasy) [reply openapi/operation.go:172]: https://github.com/speakeasy-api/openapi/discussions/68

File: yml/config.go (hunk 5 above)
Comment (TristanSpeakEasy) [reply yml/config.go]: fixed

File: json/json.go (hunk 3 above)
Comment (TristanSpeakEasy) [reply json/json.go]: fixed
